libraries/net/node.cpp

```diff
@@ -3135,7 +3155,11 @@ namespace graphene { namespace net { namespace detail {
       {
         if (firewall_check_state->expected_node_id != peer->node_id && // it's not the node who is asking us to test
             !peer->firewall_check_state && // the peer isn't already performing a check for another node
-            firewall_check_state->nodes_already_tested.find(peer->node_id) == firewall_check_state->nodes_already_tested.end() &&
+            _address_builder != nullptr &&
+            peer->get_remote_endpoint().valid() && 
+            _address_builder->should_advertise(*peer->get_remote_endpoint()) && // can advertise who is about to be asked
+            firewall_check_state->nodes_already_tested.find(peer->node_id) 
+                  == firewall_check_state->nodes_already_tested.end() && // we haven't already asked
             peer->core_protocol_version >= 106)
         {
           wlog("forwarding firewall check for node ${to_check} to peer ${checker}",
```

Comment:
I think the log level `warn` here is a bit too high. Never noticed it in the log files though, don't know why.